hedera/api.py

```diff
@@ -310,19 +320,29 @@ def get_data(self):
             })
         return lang_list
 
+    def check_data(self, data):
+        keys = ["familiarity", "headword", "gloss", "vocabulary_list_id"]
+        for key in keys:
+            if key not in data:
+                return False
+        return True
+
     def post(self, request, *args, **kwargs):
         data = json.loads(request.body)
-        _, created = PersonalVocabularyListEntry.objects.get_or_create(**data)
-        return JsonResponse({"data": {"created": created}})
-
-# TODO add suggested node functionality
-# class LatticeNodesAPI(APIView):
-
-#     def get_data(self):
-#         headword = self.request.GET.get("headword")
-#         filtered_headword_iterable = filter(str.isalnum, headword)
-#         filtered_headword_string = "".join(filtered_headword_iterable)
-#         qs = LatticeNode.objects.filter(label__iregex=r"\y"+ re.escape(filtered_headword_string) + r"\y")
-#         data = serializers.serialize('json', qs)
-#         json_data = json.loads(data)
-#         return json_data
+        checked_data = self.check_data(data)
+        if checked_data is not True:
+            return HttpResponseBadRequest("Missing required fields")
+        if "node" in data:
+            data["node"] = get_object_or_404(LatticeNode, pk=data["node"])
+        new_entry = PersonalVocabularyListEntry.objects.create(**data)
+        return JsonResponse({"data": {"created": True, "data": new_entry.data()}})
+
+
+class LatticeNodesAPI(APIView):
+
+    def get_data(self):
+        headword = self.request.GET.get("headword")
+        filtered_headword_iterable = filter(str.isalnum, headword)
+        filtered_headword_string = "".join(filtered_headword_iterable)
+        qs = LatticeNode.objects.filter(label__iregex=r"\y" + re.escape(filtered_headword_string) + r"\y")
```

Comment:
Just curious, but what is `\y` in the regex?

Reply:
Think it might have something to do with how Postgres uses word boundaries. I found an old stackoverflow post here [PostgreSQL Regex Word Boundaries](https://stackoverflow.com/questions/3825676/postgresql-regex-word-boundaries/29798772). We might have to tweak this to accommodate the changes Ivy was talking about.

Reply:
Should we hold off on refactoring the regex or refactor the regex to account for numerals in the headword?

Reply:
@jtauber We had a question about the following when looking up headwords in relation to the personal vocabulary list:
- if we wanted to search for a headword should we look at `lattices_latticenodes`  table
- how are the headword stored based on multiple meanings between the same headword